std/typecons.d

```diff
@@ -6548,7 +6551,7 @@ to deallocate the corresponding resource.
             return;
         // Done, destroy and deallocate
         .destroy(_refCounted._store._payload);
-        _refCounted.deallocateStore();
+        () @trusted { _refCounted.deallocateStore(); }();
```

Comment:
I think it's only this line that needs to be `@system` without -dip1000. I'm not sure if there is a `version`/`enum` for `-dip1000` anywhere already, but if not:

```d
import std.traits : isSafe;
enum dip1000Enabled = isSafe!((int x) => *&x);
```